### DISCUSSION TITLE
GPU capture optional plugin

### DISCUSSION THREAD

**Discussion by CedricGuillemet** [BabylonJS/BabylonNative#862] [Development]
Metal and bgfx (with RenderDoc Android/Windows/Linux, Xcode on Mac/iOS) provide API to do GPU frame capture programmatically.
This is needed when a user finds a bug in rendering on a device we don't have or it hard to repro without all the assets/scripts.

We can create a BabylonNative plugin that will do the capture. Letting the use zip the file and upload it somewhere so we can debug it (like a crash dump but for GPU).

This plugin can (should?) be separate from the main repo and is optional. Proposed JS API is simple:

```
BabylonGPUCapture.StartCapture("NameOfCapture");
...
do rendering
...
BabylonGPUCapture.StopCapture();
```

Repo readme.md will describe ways to get temporary files from the device.

**Comment by bghgary** [comment]
Here is an example of something we did to debug XR on iOS: https://github.com/BabylonJS/BabylonNative/compare/master...bghgary:arkit-custom-capture?expand=1

Doing this generically would be great!
